### PR TITLE
Implement conversion from cow to std cow

### PR DESCRIPTION
**Pull request by Dav1dde**
Closes: #474

### CODE REVIEW

**Comment by Dav1dde** [comment]
Did some experiments with `miri` and it seems to be happy.

File: metrics/src/cow.rs [hunk 1]
```diff
@@ -178,6 +178,35 @@ where
...
+    pub fn as_borrowed(&self) -> Option<&'a T> {
+        match self.metadata.kind() {
+            Kind::Owned | Kind::Shared => None,
+            Kind::Borrowed => {
+                // SAFETY: We know the contained data is borrowed from 'a, we're simply
+                // restoring the original immutable reference and returning a copy of it.
+                Some(unsafe { &*T::borrowed_from_parts(self.ptr, &self.metadata) })
+            }
+        }
+    }
```
Comment (tobz) [inline metrics/src/cow.rs:209]: Unless there's a compelling reason to have this be a standalone method, we should just move the logic down into `impl From<Cow<'a, T>> for std::borrow::Cow<'a, T>`.

**Review by tobz — CHANGES_REQUESTED** [review]
The logic is sound, but I think we just need to clean up the ergonomics a little.

File: metrics/src/cow.rs (hunk 1 above)
Comment (Dav1dde) [reply metrics/src/cow.rs:209]: The problem with that is, the `From` impl always requires a `Clone` (or taking ownership, but the way the `SharedString` is exposed it is usually through a reference), the `as_borrowed()` allows for a decision.

**Comment by Dav1dde** [comment]
Removed the `as_borrowed` and only left the cow impl.

**Comment by tobz** [comment]
Released in `metrics@v0.23.0`.

Thanks again for your contribution. 🙏🏻

**Pull request merged**